frontend/src/toolbar/elements/heatmapToolbarMenuLogic.ts

```diff
@@ -426,8 +457,28 @@ export const heatmapToolbarMenuLogic = kea<heatmapToolbarMenuLogicType>([
                 actions.setHeatmapScrollY(scrollY)
             }
         }, 100)
+
+        // we bundle the whole app with the toolbar, which means we don't need ES5 support
+        // so we can use IntersectionObserver
+        // eslint-disable-next-line compat/compat
+        const intersectionObserver = new IntersectionObserver((entries) => {
+            entries.forEach((entry) => {
+                const element = entry.target as HTMLElement
+                const rect = element.getBoundingClientRect()
+                actions.updateElementMetrics(element, entry.isIntersecting, rect)
+            })
+        })
+
+        // Store for cleanup
+        cache.intersectionObserver = intersectionObserver
     }),
     beforeUnmount(({ cache }) => {
         clearInterval(cache.scrollCheckTimer)
+        cache.intersectionObserver?.disconnect()
```

Comment:
Does it make sense to also clear `elementMetrics` map here, or will Kea handle it?

Reply:
i think we reset elsewhere... but the on/off state of the toolbar is overly complicated still